#!/usr/bin/env python
PACKAGE = "robospect_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# PLANNER PARAMS
gen.add("d_lookahead_min",    double_t,    0, "Lookahead min", 3.0,  0.5, 5.0)
gen.add("d_lookahead_max", double_t, 0, "Lookahead max",   3.5 , 1.0, 5.5)
gen.add("desired_distance", double_t, 0, "Desired distance",   2.0 , 0.5, 4.0)
gen.add("kr", double_t, 0, "Kr constant",   0.8 , 0.1, 1.0)

# TUNNEL PARAMS
gen.add("footprint_width",    double_t,    1, "Footprint width", 0.6,  0.1, 2.0)
gen.add("footprint_length", double_t, 1, "Footprint length",   0.7 , 0.1, 3.0)
gen.add("crane_length", double_t, 1, "Crane length", 3.0, 0.1, 5.0)
gen.add("lateral_clearance", double_t, 1, "Lateral clearance",   0.2 , 0.1, 1.0)
gen.add("obstacle_range", double_t, 1, "Obstacle range",   0.4 , 0.1, 2.0)
gen.add("lookahead_distance", double_t, 1, "Lookahead distance",   2.0 , 0.1, 3.0)
gen.add("preferred_wall_side", str_t, 1, "Preferred wall side",  "RIGHT")
gen.add("ransac_threshold", double_t, 1, "Ransac threshold",   0.05 , 0.01, 0.2)
gen.add("max_y_dist", double_t, 1, "Max y dist",   5.0 , 1.0, 10.0)

#gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
#gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

exit(gen.generate(PACKAGE, "robospect_planner", "Planner"))
